packages/ai/src/generate-object/stream-object.ts

```diff
@@ -448,11 +482,15 @@ class DefaultStreamObjectResult<PARTIAL, RESULT, ELEMENT_STREAM>
       ObjectStreamPart<PARTIAL>,
       ObjectStreamPart<PARTIAL>
     >({
-      transform(chunk, controller) {
+      async transform(chunk, controller) {
         controller.enqueue(chunk);
 
+        if (chunk.type === 'object' || chunk.type === 'text-delta') {
+          await onChunk?.({ chunk });
+        }
+
         if (chunk.type === 'error') {
-          onError({ error: wrapGatewayError(chunk.error) });
+          await onError({ error: wrapGatewayError(chunk.error) });
```

Comment:
await was missing here from onError, which returns a promise and is also awaited in streamText.